### ISSUE TITLE
add flag that retries a release with `--force`

### ISSUE THREAD

**Issue by Morriz** [roboll/helmfile#1249]
Helmfile now stops when it encounters helm throwing a k8s error of type forbidden. This almost always means that a resource can't be patched and should be recreated. It would be enormously helpful if helmfile can detect these errors and retries with `--force=true`, but only with an opt-in flag (like `retryWithForce`?) set to false.

@mumoshu what do you think?

**Comment by mumoshu** [comment]
@Morriz Hmm, yeah this sounds great. But how would Helmfile actually determine if an error is an user-error/permanent error or a transient error?

**Comment by Morriz** [comment]
Maybe a precondition can be that the error has the word “forbidden” in it? (I have no clue of kubectl’s error code system.)

Anyway, retrying with `—force` is not more destructive than the flag name implies, so it could just rerun it on any error even. I just think the above precondition check is a bit better (even though it’s not congruent with kubectl).

**Comment by mumoshu** [comment]
@Morriz Gotcha. How many times do you want a failed attempt to be retried? Only once?

**Comment by Morriz** [comment]
Yeah, it would suffice to do just the retry with `--force`. If it would be about eventual consistency then it would be nicer to try without force at least once more, before trying with it. That warrants another flag maybe (like `retryCount` defaulting to 0)? To have an operation retry to overcome network issues (I have these a lot with azure).
